### PR TITLE
Add docs to simplejson and sin

### LINKED ISSUE #2011: Document `simpleJSONXYZ` functions.
Create documentation for the `simpleJSONXYZ` function. See #1833 for details on how to complete this issue.

- [x] simpleJSONExtractBool
- [x] simpleJSONExtractFloat
- [x] simpleJSONExtractInt
- [x] simpleJSONExtractRaw
- [x] simpleJSONExtractString
- [x] simpleJSONExtractUInt
- [x] simpleJSONHas

Child issue of #1833.

### LINKED ISSUE #2012: Document `sin` function.
Create documentation for the `sin` function. See #1833 for details on how to complete this issue.

Child issue of #1833.

### LINKED ISSUE #2038: Document `visitParamXYZ` functions.
Create documentation for the `visitParamXYZ` functions. See #1833 for details on how to complete this issue.

- [ ] visitParamExtractBool
- [ ] visitParamExtractFloat
- [ ] visitParamExtractInt
- [ ] visitParamExtractRaw
- [ ] visitParamExtractString
- [ ] visitParamExtractUInt
- [ ] visitParamHas

Child issue of #1833.

### PR DESCRIPTION
**Pull request by antaljanosbenjamin**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add documentation to `simpleJSON` functions, unify documentation for `visitParam` functions and add example to `sin`.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 981c507d8007a4f7761a83a2ecfa0956a364317d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60881/981c507d8007a4f7761a83a2ecfa0956a364317d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by alexey-milovidov** [comment]
Let's also copy to inline docs...

File: src/Common/FunctionDocumentation.cpp [hunk 1]
```diff
@@ -36,6 +36,7 @@ std::string FunctionDocumentation::categoriesAsString() const
 
     auto it = categories.begin();
     std::string res = *it;
+    ++it;
```
Comment (rschu1ze) [inline src/Common/FunctionDocumentation.cpp:39]: Maybe replace l. 34-42 by  `return boost::algorithms::join(categories, ", ")`?

**Review by rschu1ze — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
Test failures are not related.

**Pull request merged**